### ISSUE TITLE
Count function returns incorrect results with allow filtering.

### ISSUE THREAD

**Issue by gnumoreno** [scylladb/scylla#6056]
*Installation details*
Scylla version (or git commit hash): 3.3.0-202003200707
Cluster size: 1 node
OS (RHEL/CentOS/Ubuntu/AWS AMI): docker

```
cqlsh> select * from pandemic.covid_world_deaths_by_country_date where country = 'Brazil' and deaths > 1 ALLOW FILTERING;

 country | date       | deaths | latitude | longitude | state
---------+------------+--------+----------+-----------+-------
  Brazil | 2020-03-20 |     11 |  -14.235 |  -51.9253 |  null
  Brazil | 2020-03-19 |      6 |  -14.235 |  -51.9253 |  null
  Brazil | 2020-03-18 |      3 |  -14.235 |  -51.9253 |  null

---MORE---
(3 rows)
```

```
cqlsh> select count(*) from pandemic.covid_world_deaths_by_country_date where country = 'Brazil' and deaths > 1 ALLOW FILTERING;

 count
-------
    59

(1 rows)
```

Related to #6055 ?

### Docker Reproducer
NOTE: gtihub does not let me attach .sh files so renamed to .txt
[db_setup.txt](https://github.com/scylladb/scylla/files/4364105/db_setup.txt)
[docker_setup.txt](https://github.com/scylladb/scylla/files/4364106/docker_setup.txt)

```
mv docker_setup.txt docker_setup.sh
mv db_setup.txt db_setup.sh
chmod +x docker_setup.sh
./docker_setup.sh
select * from pandemic.covid_world_deaths_by_country_date where country = 'Brazil' and deaths > 1 ALLOW FILTERING;
select count(*) from pandemic.covid_world_deaths_by_country_date where country = 'Brazil' and deaths > 1 ALLOW FILTERING;
```

PS.: Pardon my horrible programming.
@psarna

**Comment by psarna** [comment]
Related to https://github.com/scylladb/scylla/issues/5708, fixed by https://github.com/scylladb/scylla/pull/6034

**Comment by psarna** [comment]
@gnumoreno I verified your scripts on master with the fix from #6034 already applied, and it returns correct `count(*)`.

**Comment by gnumoreno** [comment]
@psarna thanks!
